keel-core/src/main/kotlin/com/netflix/spinnaker/keel/validators/DeliveryConfigProcessor.kt

```diff
@@ -0,0 +1,18 @@
+package com.netflix.spinnaker.keel.validators
+
+import com.netflix.spinnaker.keel.core.api.SubmittedDeliveryConfig
+
+/**
+ * A component that can validate or transform a [SubmittedDeliveryConfig] before it is persisted.
+ * For example, applying default values, or ensuring semantic correctness.
+ */
+interface DeliveryConfigProcessor : (SubmittedDeliveryConfig) -> SubmittedDeliveryConfig
```

Comment:
This is a new interface for applying changes to, or validating the semantics of a `SubmittedDeliveryConfig`. I propose that in a future PR we break up `DeliveryConfigValidator` into implementations of this interface.

Reply:
The implementation of this that defaults the service account based on the application owner email (checking against Pandora that it's a group email) will be in `keel-nflx`